Saleh alsadaoon

NotHacker101 opened the pull request:
Done adding user stories and changed in the code to add the three features.

Calinou committed:
Add instructions on importing all projects at once in the README

This may be helpful to newcomers; see #423.

aaronfranke committed:
Merge pull request #424 from Calinou/readme-add-scan-instructions

Add instructions on importing all projects at once in the README

aaronfranke committed:
2.5D Demo Project for Mono C#

Co-authored-by: Stephen Agbete (Steph155) <bgsteph15@mail.com>

aaronfranke committed:
2.5D Demo Project for GDScript

Co-authored-by: Stephen Agbete (Steph155) <bgsteph15@mail.com>

aaronfranke committed:
2.5D Editor Viewport for GDScript

aaronfranke committed:
2.5D Editor Viewport for Mono C#

aaronfranke committed:
Add screenshots and tweak README files

aaronfranke committed:
Add music and set physics engine to Bullet

aaronfranke committed:
Add icon for the 2.5D editor viewport

aaronfranke committed:
Merge pull request #315 from aaronfranke/2point5d

Add 2.5D Demo Projects

aaronfranke committed:
Update C# demos to Godot 3.2

aaronfranke committed:
Convert most demos to GLES 2

Also fix Viewport error in 3D scaling demo, add an icon to Multiplayer Bomber, and rename BPM sync files.

aaronfranke committed:
Open and save 3D IK demo in Godot 3.2

aaronfranke committed:
Refactor the Joypads demo

aaronfranke committed:
Merge pull request #426 from aaronfranke/gles2

Convert most demos to GLES 2 and update to Godot 3.2

aaronfranke committed:
[2D Platformer] Move Player out of Level, set limits in a script

aaronfranke committed:
[2D Platformer] Add a splitscreen mode

aaronfranke committed:
Merge pull request #427 from aaronfranke/joypad

Refactor the Joypads demo

aaronfranke committed:
Remove grid based movement demo

I don't think there's a point to this since we also have the JRPG demo.

aaronfranke committed:
Merge pull request #428 from aaronfranke/splitscreen

Add a splitscreen mode to the 2D platformer.

aaronfranke committed:
Add README files to 2D demos

aaronfranke committed:
Add README files to 3D demos

aaronfranke committed:
Add README files to demos with multiple versions

aaronfranke committed:
Add README files to Viewport demos

aaronfranke committed:
Add README to and fix Circle Pop demo

Circle Pop gets its own commit since I found a bug that had to be fixed

aaronfranke committed:
Add README files to audio demos

aaronfranke committed:
Merge pull request #431 from aaronfranke/no-grid

Remove the grid based movement demo

aaronfranke committed:
Add README files to GUI demos

aaronfranke committed:
Add README files to loading and multiplayer demos

aaronfranke committed:
Add README files to most of the misc demos

The Android ones can wait for later

Calinou committed:
Fix incorrect comment in hierarchical FSM demo script

This closes #433.

aaronfranke committed:
Merge pull request #432 from aaronfranke/readme

Add README files to many demos

aaronfranke committed:
Merge pull request #435 from Calinou/fix-hfsm-comment

Fix incorrect comment in hierarchical FSM demo script

aaronfranke committed:
Some changes to Astar code

aaronfranke committed:
Some changes to kinematic character 3D

aaronfranke committed:
Merge pull request #436 from aaronfranke/astar-kinematic

Some changes to Astar and Kinematic Character 3D

aaronfranke committed:
Improve the issue template and add a PR template

Improve the comments in the issue template, and the PR template is entirely one comment.

aaronfranke committed:
Merge pull request #441 from aaronfranke/issue-template

Improve the issue template and add a PR template

aaronfranke committed:
Fix C# project files for Dodge the Creeps

aaronfranke committed:
Merge pull request #442 from aaronfranke/dodge-the-csproj

Fix C# project files for Dodge the Creeps

aaronfranke committed:
Tweak 2D navigation polygon demo camera and remove RPG unused files

aaronfranke committed:
Merge pull request #443 from aaronfranke/2d-nav-poly-tweak

Tweak 2D navigation polygon demo camera and remove RPG unused files

aaronfranke committed:
Convert materials to tres format in Truck Town

aaronfranke committed:
Merge pull request #444 from aaronfranke/truck-town-tres

Convert materials to tres format in Truck Town

aaronfranke committed:
Organize and simplify the Kinematic Character 2D demo

aaronfranke committed:
Move Dodge the Creeps license information into the README files

aaronfranke committed:
Multitouch View tweaks

VitaZheltyakov committed:
Multiple-threads loading (#425)

Add Multiple-threads loading demo

aaronfranke committed:
Style tweaks and add import files to multiple threads loading demo

aaronfranke committed:
Merge pull request #448 from aaronfranke/mult-thread-load

Style tweaks and add import files to multiple threads loading demo

aaronfranke committed:
Merge pull request #445 from aaronfranke/kc2d

Organize and simplify the Kinematic Character 2D demo

aaronfranke committed:
Merge pull request #446 from aaronfranke/dodge-the-attributions

Move Dodge the Creeps license information into the README files

aaronfranke committed:
Merge pull request #447 from aaronfranke/multitouch

Multitouch View tweaks

aaronfranke committed:
Add descriptions to all projects with README files

These show up in the project manager when hovering over projects.

aaronfranke committed:
Merge pull request #449 from aaronfranke/description

Add descriptions to all projects with README files

aaronfranke committed:
Add a formatting script for Travis CI

aaronfranke committed:
Commit files changed by the formatting script

aaronfranke committed:
Remove duplicate issue templates

aaronfranke committed:
Update issue templates

aaronfranke committed:
Merge pull request #452 from aaronfranke/issue-templates

Remove duplicate issue templates

akien-mga committed:
Merge pull request #450 from aaronfranke/travis

Add a formatting script for local checking and for Travis CI

LaBodilsen committed:
simplifying steering to a single line with move_toward (#451)

* simplyfied the steering to a single line with lerp

Replaced:
	if steer_target < steer_angle:
		steer_angle -= STEER_SPEED * delta
		if steer_target > steer_angle:
			steer_angle = steer_target
	elif steer_target > steer_angle:
		steer_angle += STEER_SPEED * delta
		if steer_target < steer_angle:
			steer_angle = steer_target
	
	steering = steer_angle
with
	steering = lerp(steering, steer_target, STEER_SPEED)

and lowered STEER_SPEED to 0.2

* Update vehicle.gd

* added space around *

Commit:
Merge branch 'master' into Saleh-Alsadaoon

Commit:
Made to add 3 features

-Reduces the amount of gravity to make the game seems smoother.
-Added in a new skin to the player.
-Replace the old skin with the new one.
-Replaced all of the old animations with the new charcter animation.
-New bug has been introduced where the player gets double the size when moving left or right.
-Added a new power up that the player can pick up but doesnt do anything yet.

NotHacker101 committed:
Added new user stories

WilltheWys reviewed (commented):
changes look to be all good

reid-wynkoop reviewed (commented):
Why are there a bunch of commits to projects other than Platformer?

Commit:
fixed the backlog task made by saleh alsadoon

Commit:
Updated features.md file

Commit:
added some features on the readme file

Commit:
Update Jump.gd

Changed the jump boost comments to make it more clear and nicer coding, Because im taking the class as a masters class.

Commit:
reimport the powerups pics

Added all powerups pics that can be added in the feature with the correct naming.

Commit:
Changed gravity

Made gravity lower to make game feel better

Commit:
gun shoot speed

Made gun shoot faster to make it feel a bit more realistic.

Commit:
updated backlog

New feature

Commit:
flip the jump boost icon

mbagwell reviewed (commented):
There are so many changes, and you mentioned how you changed features, but what did you change specifically? also, there are still some conflicting files(
2d/platformer/src/Actors/Gun.gd
backlog.md
features.md).

mbagwell reviewed (commented):
Why are you adding so many backstories to different programs(that we aren't using)?

D-B-Cooper commented on 2d/dodge_the_creeps/fonts/LICENSE.txt:
```diff
@@ -1,94 +1,94 @@
+in Original or Modified Versions, may be sold by itself.
+
+2) Original or Modified Versions of the Font Software may be bundled,
+redistributed and/or sold with any software, provided that each copy
+contains the above copyright notice and this license. These can be
+included either as stand-alone text files, human-readable headers or
+in the appropriate machine-readable metadata fields within text or
+binary files as long as those fields can be easily viewed by the user.
+
+3) No Modified Version of the Font Software may use the Reserved Font
+Name(s) unless explicit written permission is granted by the corresponding
+Copyright Holder. This restriction only applies to the primary font name as
+presented to the users.
+
+4) The name(s) of the Copyright Holder(s) or the Author(s) of the Font
+Software shall not be used to promote, endorse or advertise any
+Modified Version, except to acknowledge the contribution(s) of the
+Copyright Holder(s) and the Author(s) or with their explicit written
+permission.
+
+5) The Font Software, modified or unmodified, in part or in whole,
+must be distributed entirely under this license, and must not be
+distributed under any other license. The requirement for fonts to
+remain under this license does not apply to any document created
+using the Font Software.
+
+TERMINATION
+This license becomes null and void if any of the above conditions are
+not met.
+
+DISCLAIMER
+THE FONT SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO ANY WARRANTIES OF
+MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT
+OF COPYRIGHT, PATENT, TRADEMARK, OR OTHER RIGHT. IN NO EVENT SHALL THE
+COPYRIGHT HOLDER BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+INCLUDING ANY GENERAL, SPECIAL, INDIRECT, INCIDENTAL, OR CONSEQUENTIAL
+DAMAGES, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+FROM, OUT OF THE USE OR INABILITY TO USE THE FONT SOFTWARE OR FROM
+OTHER DEALINGS IN THE FONT SOFTWARE.
```
Is there a reason this is marked as changed? You may have some settings off in your line endings, may want to look into that for future merges.

D-B-Cooper reviewed (commented):
You may have a setting off for your line endings, please see my comment in the request. Otherwise it looks good.

DSF256 reviewed (commented):
Okay, so the relevant changes seem pretty good.  I haven't noticed any major graphical bugs or anything of the sort, so that's good.  I'm not too what all of the other files were modified for, though.  I don't think our 2D platformer subfolder even has 467 files in it.  That might make merging this a bit of a challenge.
